ui/src/config/section/storage.js

```diff
@@ -223,7 +223,7 @@ export default {
           label: 'label.change.offering.for.volume',
           args: ['id', 'diskofferingid', 'size', 'miniops', 'maxiops', 'automigrate'],
           dataView: true,
-          show: (record, store) => { return ['Allocated', 'Ready'].includes(record.state) && ['Admin'].includes(store.userInfo.roletype) },
+          show: (record, store) => { return ['Allocated', 'Ready'].includes(record.state) && (['Admin'].includes(store.userInfo.roletype) || !!store.apis.changeOfferingForVolume) },
```

Comment:
is the double exclamation mark an error or some feature I don't know about? It would seem to me either one
```suggestion
          show: (record, store) => { return ['Allocated', 'Ready'].includes(record.state) && (['Admin'].includes(store.userInfo.roletype) || !store.apis.changeOfferingForVolume) },
```
 or both
```suggestion
          show: (record, store) => { return ['Allocated', 'Ready'].includes(record.state) && (['Admin'].includes(store.userInfo.roletype) || store.apis.changeOfferingForVolume) },
```
 should be removed

Reply:
Hi @DaanHoogland ,
The “!!” operators are no big deal, they are a double negation. I did it to force the return of a boolean, but I really don't think it makes any difference in this case

https://stackoverflow.com/questions/10467475/double-negation-in-javascript-what-is-the-purpose

Reply:
I see @LayonRibeiro , thanks for this little insider knowledge. I'd rather see a `(´changeOfferingForVolume´ in this.$store.getters.apis)` as @weizhouapache describes. It is not as concise but very clear for future maintainers.